tests/src/Kernel/AccessTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_contact_forms\Kernel;
+
+use Drupal\contact\Entity\ContactForm;
+use Drupal\Core\Access\AccessResult;
+
+/**
+ * Test the contact form access control.
+ */
+class AccessTest extends ContactFormTestBase {
+
+  /**
+   * Tests corporate contact form access.
+   */
+  public function testCorporateFormAccess(): void {
+    $contact_form = ContactForm::create(['id' => 'oe_contact_form']);
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'is_corporate_form', TRUE);
+    $contact_form->save();
+
+    // Assert user with "access corporate contact form" permission has access
+    // to contact_form entity.
+    $account = $this->createUser(['name' => 'test1'], ['access corporate contact form']);
+    \Drupal::currentUser()->setAccount($account);
+    $this->assertTrue($contact_form->access('view', \Drupal::currentUser()));
+    // Assert user without "access corporate contact form" permission does not.
+    $account = $this->createUser(['name' => 'test2'], ['access site-wide contact form']);
+    \Drupal::currentUser()->setAccount($account);
+    $this->assertFalse($contact_form->access('view', \Drupal::currentUser()));
+
+    // Test access check service.
+    $view_access = \Drupal::service('oe_contact_forms.access_checker');
+    // Assert form with allow_canonical_url is allowed.
+    $contact_form = ContactForm::create(['id' => 'canonical_allowed']);
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'is_corporate_form', TRUE);
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'allow_canonical_url', TRUE);
+    $this->assertEquals($view_access->access($contact_form), AccessResult::allowed());
+    // Assert form without allow_canonical_url is forbidden.
+    $contact_form = ContactForm::create(['id' => 'canonical_not_allowed_no_alias']);
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'is_corporate_form', TRUE);
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'allow_canonical_url', FALSE);
+    $this->assertEquals($view_access->access($contact_form), AccessResult::forbidden());
+  }
+
+  /**
+   * Tests default contact form access.
+   */
+  public function testDefaultFormAccess(): void {
```

Comment:
Don't make another method for this. Just change the form in the other method from corporate to non corporate and test again.

Reply:
merged